tests/tasks/test_refresh_schemas.py

```diff
@@ -188,3 +189,90 @@ def test_refresh_schema_update_column(self):
         column_metadata = ColumnMetadata.query.all()
         self.assertNotEqual(column_metadata[0].to_dict(), self.EXPECTED_COLUMN_METADATA)
         self.assertEqual(column_metadata[0].to_dict()['type'], UPDATED_COLUMN_TYPE)
+
+    def test_refresh_schema_get_all_table_samples_rate_limits(self):
+        NEW_COLUMN_NAME = 'new_column'
+        NUM_TABLES = 105
+        tables = []
+
+        for i in range(NUM_TABLES):
+            tables.append({
+                'name': 'table{}'.format(i),
+                'columns': [NEW_COLUMN_NAME],
+                'metadata': [{
+                    'name': NEW_COLUMN_NAME,
+                    'type': self.COLUMN_TYPE,
+                }]
+            })
+
+        self.patched_get_schema.return_value = tables
+        self.factory.data_source.query_runner.configuration['samples'] = True
+
+        refresh_schema(self.factory.data_source.id)
+        get_all_table_samples(self.factory.data_source.id)
+
+        # There's a total of 105 tables
+        table_metadata = TableMetadata.query.all()
+        self.assertEqual(len(table_metadata), NUM_TABLES)
```

Comment:
No need to load all data here with `all()`, just use `count()` instead for all the count tests.